File: PHP/Patterns/src/Behavioral/Observer/Models/Customer.php
<?php
namespace DesignPatterns\Behavioral\Observer\Models;

use DesignPatterns\Behavioral\Observer\Interfaces\Subscriber;
use Override;

class Customer implements Subscriber
{

    /**
     * @var string
     */
    private string $name;

    /**
     * @param string $name
     */
    public function __construct(string $name)
    {
        $this->name = $name;
    }

    /**
     * @return string
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * @param string $message
     * @return void
     */
    #[Override] public function notify(string $message): void
    { 
        echo 'Notifying ' . $this->name . ' about ' . $message . PHP_EOL;
    }

    
}

File: Python/observer/Customer.py

from Python.observer.Subscriber import Subscriber


class Customer(Subscriber):
    def __init__(self, name: str, email: str):
        self.email = email
        self.name = name


    def get_name(self) -> str:
        return self.name

    def get_email(self) -> str:
        return self.email


    def notify(self, message):
            print(f"Notifying {self.name} about {message}")
